syntax = "proto3";

package proto;

enum Tetro {
    TETRO_UNSPECIFIED = 0;
    TETRO_I = 1;
    TETRO_O = 2;
    TETRO_T = 3;
    TETRO_S = 4;
    TETRO_Z = 5;
    TETRO_J = 6;
    TETRO_L = 7;
}

message Int32Array{
    repeated int32 value = 1;
}

message TetroMessage {
    Tetro tetro = 1;
    repeated Int32Array data = 2;
}

message TetroPositionMessage {
    int32 col = 1;
    int32 row = 2;
}

message TetroHeapMessage {
    repeated Int32Array heap = 1;
}